src/python_testing/TC_OpstateCommon.py

```diff
@@ -0,0 +1,898 @@
+                    asserts.fail("The list shall include structs with the following OperationalStateIds: %s", defined_states)
+
+        # STEP 4: TH sends Start command to the DUT
+        self.step(4)
+        if self.pics_guard(self.check_pics("%s.S.C02.Rsp" % self.test_info.pics_code) and
+                           self.check_pics("%s.S.C04.Tx" % self.test_info.pics_code)):
+            await self.send_cmd_expect_response(endpoint=endpoint,
+                                                cmd=commands.Start(),
+                                                expected_response=cluster.Enums.ErrorStateEnum.kNoError)
+
+        # STEP 5: TH reads from the DUT the OperationalState attribute
+        self.step(5)
+        if self.pics_guard(self.check_pics("%s.S.A0004" % self.test_info.pics_code)):
+            await self.read_and_expect_value(endpoint=endpoint,
+                                             attribute=attributes.OperationalState,
+                                             expected_value=cluster.Enums.OperationalStateEnum.kRunning)
+
+        # STEP 6: TH reads from the DUT the OperationalError attribute
+        self.step(6)
+        if self.pics_guard(self.check_pics("%s.S.A0005" % self.test_info.pics_code)):
+            await self.read_and_expect_property_value(endpoint=endpoint,
+                                                      attribute=attributes.OperationalError,
+                                                      attr_property="errorStateID",
+                                                      expected_value=cluster.Enums.ErrorStateEnum.kNoError)
+
+        # STEP 7: TH reads from the DUT the CountdownTime attribute
+        self.step(7)
+        if self.pics_guard(self.check_pics("%s.S.A0002" % self.test_info.pics_code)):
+            initial_countdown_time = await self.read_expect_success(endpoint=endpoint,
+                                                                    attribute=attributes.CountdownTime)
+            if initial_countdown_time is not NullValue:
+                asserts.assert_true(0 <= initial_countdown_time <= 259200,
+                                    "CountdownTime(%s) must be between 0 and 259200" % initial_countdown_time)
+
+        # STEP 8: TH reads from the DUT the PhaseList attribute
+        self.step(8)
+        if self.pics_guard(self.check_pics("%s.S.A0000" % self.test_info.pics_code)):
+            phase_list = await self.read_expect_success(endpoint=endpoint,
+                                                        attribute=attributes.PhaseList)
+            phase_list_len = 0
+            if phase_list is not NullValue:
+                phase_list_len = len(phase_list)
+                asserts.assert_less_equal(phase_list_len, 32,
+                                          "PhaseList length(%d) must be less than 32!" % phase_list_len)
+
+        # STEP 9: TH reads from the DUT the CurrentPhase attribute
+        self.step(9)
```

Comment:
Per the TC, step 9 should be skipped if the phase_list is NULL.

Reply:
if phase_list is NULL, the variable `phase_list_len` is going to be 0, which sets the pics_guard condition to False, and forces the step to be skipped.